tests/structured_conf/test_structured_config.py

```diff
@@ -691,6 +691,29 @@ def test_promote_to_object(self, class_type: str) -> None:
         assert OmegaConf.get_type(conf) == module.BoolConfig
         assert conf.with_default is False
 
+    def test_set_key_with_with_dataclass(self, class_type: str) -> None:
+        module: Any = import_module(class_type)
+        cfg = OmegaConf.create({"foo": [1, 2]})
+        cfg.foo = module.ListClass()
```

Comment:
Look at test_assignment_to_nested_structured_config, I think it's testing the same thing.

Reply:
Ok.

Reply:
It's not, here ListClass is assigned in an Any ref_type key, in test_assignment_to_nested_structured_config Nested is assigned in a Nested ref_type key.
Anyways, I added that test because there was a part of basecontainer that wasn't covered and, this test completed the coverage.

Reply:
If the code coverage said it was not covered than it was not covered.